multinet/util.py

```diff
@@ -4,14 +4,39 @@
 
 from flask import Response
 
-from typing import Sequence, Any, Generator
+from typing import Sequence, Any, Generator, Dict, Set
 
 from . import db
 from .errors import DatabaseNotLive, DecodeFailed
+from .types import EdgeTableProperties
 
 TEST_DATA_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../test/data"))
 
 
+def get_edge_table_properties(workspace: str, edge_table: str) -> EdgeTableProperties:
+    """Return extracted information about an edge table."""
+
+    loaded_workspace = db.db(workspace)
+    edges = list(loaded_workspace.collection(edge_table).all())
```

Comment:
Instead of coercing to a list here, it would be safer to treat this as an iterable sequence. I don't think you need the whole list in the for-loop in line 26 below; correct me if I'm wrong.